src/gentypes.jl

```diff
@@ -477,17 +477,23 @@ function typecode(rosname::String, super::Symbol, members::Vector)
     #(4) Convert from PyObject
     push!(exprs, :(
         function convert(jlt::Type{$jlsym}, o::PyObject)
-            if convert(String, o["_type"]) != _typerepr(jlt)
+            if convert(String, o."_type") != _typerepr(jlt)
                 throw(InexactError(:convert, $jlsym, o))
             end
             jl = $jlsym()
             #Generated code here
             jl
         end
     ))
-    #(5) Accessing member variables through getindex
+    #(5) Accessing member variables through getproperty
     push!(exprs, :(
-        getindex(::Type{$jlsym}, s::Symbol) = RobotOS._rospy_objects[$rosname][s]
+        getproperty(::Type{$jlsym}, s::Symbol) = begin
```

Comment:
This implementation prioritizes resolution of `getproperty` in `PyObject` world before falling back to the default implementation for Julia types (i.e., `getfield`). I think the other way around in terms of prioritization could also reasonable; in any case I suppose it's possible that there could be a collision some day (e.g., if `s = :name` or any other default field of Julia types), but I think it's pretty unlikely and not worth giving up the ergonomic benefits of this implementation. That is, I prefer this approach as it feels so much nicer just to write `Marker.CUBE` over alternatives:

1. `Marker."CUBE"` (i.e., defining `getproperty(::Type{$jlsym}, s::String)` here)
2. `Marker[:CUBE]` (sticking with the previous `getindex` method)

Reply:
I'd be inclined to replace the fallback with an error message that directs the user to `getfield` if they weren't trying to access a message constant. It will probably be much more common that someone has just misspelled the constant they want and they would probably prefer an more useful error message than just being told that `DataType` doesn't have the field they tried to get.

Reply:
Thanks for the review! Unfortunately a lot of the Julia type introspection utilities (e.g., `fieldnames` tested here: https://github.com/jdlangs/RobotOS.jl/blob/master/test/typegeneration.jl#L87, implemented here: https://github.com/JuliaLang/julia/blob/e449cd4ab0f5c00cb8c7ac5754dc2873c42199ee/base/reflection.jl#L121) use dot syntax (`getproperty`) for `getfield` to access type information, so maintaining the fallback is essential I think. I see three options:
1) Silent fallback (the original behavior of this PR).
2) Fallback with another try/catch around `getfield` to print a more informative error message as you suggest.
3) Go back to `getindex` or some other access style for message constants.

I've implemented/pushed option 2 to this PR; see an example error message below:
```julia
julia> using RobotOS

julia> @rosimport visualization_msgs.msg: Marker

julia> rostypegen()

julia> import .visualization_msgs.msg: Marker

julia> Marker.CUBE
1

julia> Marker.CUBA
ERROR: Message type 'Marker' has no property 'CUBA'.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] getproperty(::Type{Marker}, ::Symbol) at /home/schmrlng/.julia/dev/RobotOS/src/gentypes.jl:497 
```

Reply:
Thanks for the extra info; option 2 works for fine for me.